### PR TITLE
KEP 3458 - promote to stable

### PR DESCRIPTION
**Pull request by alexanderConstantinescu**
#### What type of PR is this?

/kind feature 

#### What this PR does / why we need it:

Promotes the feature `StableLoadBalancerNodeSet` to GA following https://github.com/kubernetes/enhancements/pull/4424

/cc @thockin @aojea 
/sig network 
/sig cloud-provider 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Promote feature gate StableLoadBalancerNodeSet to GA
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes/enhancements/issues/3458
```

### CODE REVIEW

**Comment by sftim** [comment]
Would this need a changelog entry?

File: pkg/features/kube_features.go [hunk 1]
```diff
@@ -1112,7 +1112,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	SizeMemoryBackedVolumes: {Default: true, PreRelease: featuregate.Beta},
 
-	StableLoadBalancerNodeSet: {Default: true, PreRelease: featuregate.Beta},
+	StableLoadBalancerNodeSet: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // GA in 1.30, remove in 1.31
```
Comment (aojea) [inline pkg/features/kube_features.go:1115]: You have to update line 750 too

File: staging/src/k8s.io/controller-manager/pkg/features/kube_features.go [hunk 2]
```diff
@@ -64,5 +64,5 @@ func SetupCurrentKubernetesSpecificFeatureGates(featuregates featuregate.Mutable
 var cloudPublicFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
 	CloudControllerManagerWebhook: {Default: false, PreRelease: featuregate.Alpha},
 	CloudDualStackNodeIPs:         {Default: true, PreRelease: featuregate.Beta},
-	StableLoadBalancerNodeSet:     {Default: true, PreRelease: featuregate.Beta},
+	StableLoadBalancerNodeSet:     {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // GA in 1.30, remove in 1.31
```
Comment (aojea) [inline staging/src/k8s.io/controller-manager/pkg/features/kube_features.go:67]: update line 51 too

File: pkg/features/kube_features.go (hunk 1 above)
Comment (alexanderConstantinescu) [reply pkg/features/kube_features.go:1115]: Isn't that indicating when the feature gate was promoted to beta?

Comment (thockin) [reply pkg/features/kube_features.go:1115]: It's the record of which releases had which stability

Comment (alexanderConstantinescu) [reply pkg/features/kube_features.go:1116]: OK, updated the PR and added a line indicating that GA is starting 1.30

**Comment by thockin** [comment]
Yes, this should have a release note.

Thanks!

/lgtm
/approve

**Comment by pacoxu** [comment]
/hold
UT in https://github.com/kubernetes/kubernetes/blob/c6064489223862fe1888fcbe0656ab1087461adb/staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go#L467 needs an update.

BTW, the release note needs to be added. Make a simple one. 

/release-note-edit
```release-note
Promote feature gate StableLoadBalancerNodeSet to GA
```

**Comment by sftim** [comment]
Changelog suggestion
```diff
-Promote feature gate StableLoadBalancerNodeSet to GA
+Promoted feature gate `StableLoadBalancerNodeSet` to GA
```

Please also submit a docs update PR - see https://kubernetes.io/docs/contribute/new-content/new-features/#ready-for-review-feature-gates

**Pull request merged**